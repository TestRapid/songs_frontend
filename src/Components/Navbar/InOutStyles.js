import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
	icon: {
		background: "linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)",
		color: "#ffffff"
	},
	logout: {
		color: "#ffffff",
		margin: "0.2em auto",
		backgroundColor: "red",
		"&:hover": {
			backgroundColor: "#cc0000"
		}
	},
	btn: {
		background: "linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)",
		border: 0,
		borderRadius: 3,
		boxShadow: "0 3px 5px 2px rgba(255, 105, 135, .3)",
		color: "white"
	},
	liItem: {
		display: "flex",
		justifyContent: "space-between"
	},
	liItemLink: {
		flex: "8"
	},
	liItemDelete: {
		flex: "2"
	}
}));
export default useStyles;
