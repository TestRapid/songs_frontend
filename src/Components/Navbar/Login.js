import { Badge, IconButton, Menu, MenuItem } from "@material-ui/core";
import { Delete, People } from "@material-ui/icons";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import swal from "sweetalert";

// importing styles
import useStyles from "./InOutStyles";

// importing actions
import { logout, remove_list } from "../../Store/actions/auth";

// Login component
export const Login = ({ click }) => {
	const dispatch = useDispatch();
	const classes = useStyles();
	const songs = useSelector((state) => state);

	const [anchorEl, setAnchorEl] = useState(null);
	const open = Boolean(anchorEl);

	const handleClick = (event) => {
		setAnchorEl(event.currentTarget);
	};

	const handleClose = () => {
		setAnchorEl(null);
		click();
	};

	const log_out = () => {
		dispatch(logout());
		handleClose();
	};

	const handleDelete = (e, key) => {
		e.stopPropagation();
		handleClose();
		swal({
			title: "Are you sure?",
			text: "Once song is deleted, you will not be able to recover it",
			icon: "warning",
			buttons: true,
			dangerMode: true
		}).then((res) => {
			if (res) {
				dispatch(remove_list(key, songs.auth.user));
				swal({
					title: "Deleted",
					icon: "success",
					timer: 2000
				});
			}
		});
	};

	return (
		<>
			<IconButton
				aria-label="more"
				aria-controls="long-menu"
				aria-haspopup="true"
				onClick={handleClick}
				className={classes.icon}
			>
				<People />
			</IconButton>
			<Menu
				id="long-menu"
				anchorEl={anchorEl}
				keepMounted
				open={open}
				onClose={handleClose}
				PaperProps={{
					style: {
						width: "20ch"
					}
				}}
			>
				<Link to="/profile" onClick={handleClose}>
					<MenuItem>Profile</MenuItem>
				</Link>
				<Link to="/addsong" onClick={handleClose}>
					<MenuItem>Add Song</MenuItem>
				</Link>
				<Link to="/mysongs" onClick={handleClose}>
					<MenuItem style={{ paddingTop: "10px" }}>
						<Badge
							badgeContent={
								songs.songs.list.filter(
									(l) => l.userId === songs.auth.user.id
								).length
							}
							color="error"
							showZero
							max={20}
						>
							My songs
						</Badge>
					</MenuItem>
				</Link>
				{Object.entries(songs.auth.user.playlist).map(
					([key, val], i) => (
						<MenuItem
							key={i}
							className={classes.liItem}
							style={{ paddingTop: "10px" }}
						>
							<Link
								className={classes.liItemLink}
								to={`/playlist/${key}`}
								onClick={handleClose}
							>
								<Badge
									badgeContent={val.length}
									color="error"
									showZero
									max={20}
								>
									{key}
								</Badge>
							</Link>
							{key !== "MyPlaylist" && (
								<IconButton
									className={classes.liItemDelete}
									onClick={(e) => handleDelete(e, key)}
								>
									<Delete />
								</IconButton>
							)}
						</MenuItem>
					)
				)}
				<hr></hr>
				<MenuItem className={classes.logout} onClick={log_out}>
					Logout
				</MenuItem>
			</Menu>
		</>
	);
};

export default Login;
