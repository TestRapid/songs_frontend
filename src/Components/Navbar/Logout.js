import { Grid } from "@material-ui/core";
import { Link } from "react-router-dom";

// importing custom components
import Controls from "../../Components/reuse";

// importing styles
import useStyles from "./InOutStyles";

// Logout component
export const Logout = ({ click }) => {
	const classes = useStyles();

	return (
		<>
			<Grid container spacing={1}>
				<Grid item>
					<Link
						style={{ color: "white" }}
						onClick={click}
						to="/signup"
					>
						<Controls.Button>Signup</Controls.Button>
					</Link>
				</Grid>
				<Grid item>
					<Link
						style={{ color: "white" }}
						onClick={click}
						to="/login"
					>
						<Controls.Button className={classes.btn}>
							login
						</Controls.Button>
					</Link>
				</Grid>
			</Grid>
		</>
	);
};

export default Logout;
