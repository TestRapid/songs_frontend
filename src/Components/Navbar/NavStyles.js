import { makeStyles } from "@material-ui/core";

export const useStyles = makeStyles((theme) => ({
	navbar: {
		background: `linear-gradient(90deg,	rgba(110, 94, 254) 0%, rgba(73, 63, 252, 1) 100%)`,
		height: "70px",
		display: "flex",
		justifyContent: "space-between",
		alignItems: "center",
		fontSize: "1.2em",
		padding: "0.5em 2em",
		position: "relative",
		width: "100%",
		zIndex: "10"
	},
	navbarLogo: {
		color: "#ffffff",
		cursor: "pointer",
		justifySelf: "start",
		marginLeft: "20px"
	},
	navbarToggle: {
		display: "none",
		color: "#ffffff",
		fontSize: "1.2em",
		[theme.breakpoints.down("md")]: {
			display: "block"
		}
	},
	list: {
		display: "flex",
		padding: "0.7rem 1.5rem",
		justifyContent: "center",
		alignItems: "center",
		[theme.breakpoints.down("md")]: {
			position: "absolute",
			flexDirection: "column",
			justifyContent: "start",
			alignItems: "center",
			gap: "1em",
			width: "100%",
			height: "90vh",
			backgroundColor: "#6668f4",
			top: "70px",
			left: "-100%",
			transition: "all 0.3s ease-in-out"
		},
		zIndex: "9"
	},
	listActive: {
		left: "0%"
	},
	listItem: {
		color: "#ffffff",
		padding: "0.5rem 1.5rem",
		cursor: "pointer"
	},
	listLinks: {
		color: "#ffffff",
		padding: "0.7rem 1.5rem",
		cursor: "pointer",
		textDecoration: "none",
		"&:hover": {
			backgroundColor: "#6d76f7",
			borderRadius: "4px",
			transition: "all 0.2s ease-in-out",
			[theme.breakpoints.down("md")]: {
				width: "100%",
				textAlign: "center",
				borderRadius: "0",
				backgroundColor: "#7577fa"
			}
		}
	}
}));

export default useStyles;
