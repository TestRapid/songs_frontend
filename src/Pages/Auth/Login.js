import { Paper, Grid, CircularProgress, Typography } from "@material-ui/core";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useHistory, useLocation, Link } from "react-router-dom";
import { Lock } from "@material-ui/icons";
import { useFormik } from "formik";
import * as Yup from "yup";
import swal from "sweetalert";

// input
import Control from "../../Components/reuse";

// api for authentication
import * as API from "../../Store/apis/auth";

// importing action
import { login } from "../../Store/actions/auth";

// importing useStyles
import useStyles from "./styles";

const initState = {
	mail: "",
	password: ""
};

// login page
export const Login = () => {
	const classes = useStyles();
	const history = useHistory();
	const auth = useSelector((s) => s.auth);
	const location = useLocation();
	const dispatch = useDispatch();
	const [error, setError] = useState({});
	const [isloading, setIsloading] = useState(false);
	const [validity, setValidity] = useState(false);

	const formit = useFormik({
		initialValues: initState,
		validationSchema: Yup.object({
			mail: Yup.string().required(`* required`).min(4),
			password: Yup.string().required(`* reqiured`).min(4)
		}),
		validateOnChange: validity,
		validateOnBlur: validity,
		onSubmit: (values, actions) => {
			setValidity(true);
			actions.setSubmitting(false);
			check(values, actions.setSubmitting);
		}
	});

	useEffect(() => {
		setError(formit.errors);
	}, [formit.errors]);

	const check = async (values, cb) => {
		setIsloading(true);
		try {
			const { data } = await API.getUser(values.mail);
			dispatch(login(data));
			swal({
				timer: 1000,
				title: "login success",
				buttons: false,
				icon: "success"
			});
		} catch (err) {
			setError((s) => ({
				...s,
				mail: "user not found..."
			}));
		}
		cb(true);
	};

	const { from } = location.state || { from: { pathname: "/" } };

	useEffect(() => {
		!!auth && history.replace(from);
		return () => {
			setIsloading(false);
		};
	}, [auth, from, history]);

	return (
		<Grid container>
			<Grid item xs={12} md={6} className={classes.marginAuto}>
				<Paper className={classes.paper}>
					<Grid container className={classes.ico}>
						<Grid item>
							<Lock className={classes.icon} />
							<Typography variant="h5">Login</Typography>
						</Grid>
					</Grid>
					<form onSubmit={formit.handleSubmit}>
						<Grid container spacing={2}>
							<Grid item xs={12}>
								<Control.Input
									type="email"
									name="mail"
									label="E-mail"
									value={formit.values.mail}
									onChange={formit.handleChange}
									required
									error={!!error.mail}
									helperText={error.mail}
								/>
							</Grid>
							<Grid item xs={12}>
								<Control.Input
									type="password"
									name="password"
									label="Password"
									value={formit.values.password}
									onChange={formit.handleChange}
									required
									error={!!error.password}
									helperText={error.password}
								/>
							</Grid>
							<Grid item xs={12}>
								<Control.Button
									disabled={isloading}
									fullWidth
									type="submit"
									color="primary"
									startIcon={
										isloading && (
											<CircularProgress size={25} />
										)
									}
								>
									Login
								</Control.Button>
							</Grid>
						</Grid>
					</form>
					<Grid container className={classes.toggle} spacing={2}>
						<Grid item>
							<Link to="/signup">
								<Control.Button color="default">
									Create new account...
								</Control.Button>
							</Link>
						</Grid>
					</Grid>
				</Paper>
			</Grid>
		</Grid>
	);
};
export default Login;
