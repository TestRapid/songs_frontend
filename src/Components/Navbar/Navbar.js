import { useState } from "react";
import { Link } from "react-router-dom";
import { IconButton, Typography } from "@material-ui/core";
import { Close, Menu } from "@material-ui/icons";
import { useSelector } from "react-redux";

import Logout from "./Logout";
import Login from "./Login";

// importing styles
import useStyles from "./NavStyles";

// navbar component
export const Nav = () => {
	const classes = useStyles();
	const auth = useSelector((s) => s.auth);
	const [mobile, setMobile] = useState(false);

	const handleClick = () => {
		setMobile((s) => !s);
	};

	return (
		<nav className={classes.navbar}>
			<Link to="/">
				<Typography className={classes.navbarLogo}>Feelu</Typography>
			</Link>
			<IconButton className={classes.navbarToggle} onClick={handleClick}>
				{mobile ? <Close /> : <Menu />}
			</IconButton>
			<div
				className={`${classes.list} ${!!mobile && classes.listActive}`}
			>
				<Link
					to="/songs"
					className={classes.listLinks}
					onClick={handleClick}
				>
					Songs
				</Link>
				<Link
					to="/about"
					className={classes.listLinks}
					onClick={handleClick}
				>
					About
				</Link>
				<div className={classes.listItem}>
					{!auth ? (
						<Logout click={() => setMobile(false)} />
					) : (
						<Login click={() => setMobile(false)} />
					)}
				</div>
			</div>
		</nav>
	);
};

export default Nav;
