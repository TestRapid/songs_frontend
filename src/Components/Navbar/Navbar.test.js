import { shallow } from "enzyme";

import Navbar from "./Navbar";
import configStore from "redux-mock-store";
import { Provider } from "react-redux";
import Logout from "./Logout";
import Login from "./Login";

const mockStore = configStore({
	auth: {}
});

describe("Test suite for Navbar", () => {
	let store;
	let wrapper;
	beforeEach(() => {
		store = mockStore({
			auth: {
				user: {
					firstName: "user",
					lastName: "user",
					password: "userbkj",
					gender: "male",
					mobile: "903945234",
					location: "bangalore",
					id: "user@mailo.com",
					added: 1627912422223,
					playlist: {
						MyPlaylist: []
					}
				}
			}
		});
		wrapper = shallow(
			<Provider store={store}>
				<Navbar />
			</Provider>
		);
	});

	it("snapshot test", () => {
		expect(wrapper).toMatchSnapshot();
	});

	it("test for Login component", () => {
		let login = wrapper.find(Login);
		expect(login).toBeDefined();
	});

	it("test for menu list", () => {
		let list = wrapper.find("div>Link");
		expect(list.length).toBeDefined();
	});
});

describe("test when store is has no logged in user", () => {
	let store;
	let wrapper;

	beforeEach(() => {
		store = mockStore({
			auth: null
		});
		wrapper = shallow(
			<Provider store={store}>
				<Navbar />
			</Provider>
		);
	});

	it("test for Logout component", () => {
		let logout = wrapper.find(Logout);
		expect(logout).toBeDefined();
	});
});
