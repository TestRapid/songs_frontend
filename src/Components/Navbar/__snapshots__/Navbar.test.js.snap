// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Navbar snapshot test 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Nav />
</ContextProvider>
`;
